docs/instructor-training-workflow.txt

```diff
@@ -0,0 +1,211 @@
+  - Admin selects <Training Application> marked "Pending" that has been matched with <Person>.
+  - Admin approves <Training Application>.
+  - AMY clears <Training Application>'s "Pending" flag.
+
+* [admin approves training][b]: no user record in database.
+  - Admin views <Training Application List>.
+  - Admin selects <Training Application>.
+  - Admin approves <Training Application>.
+  - AMY clears <Training Application>'s "Pending" flag.
+  - AMY creates <Person> record and associates with <Training Application>.
+
+## Administrator registers individual for training.
+
+* [admin registers user][a]: user is already in database.
+  - Administrator looks up email address or GitHub ID.
+  - Administrator => [user registers self][a].
+
+* [admin registers user][b]: user is *not* already in database.
+  - Administrator creates <Person>.
+  - Administrator => [admin registers user][a].
+
+## Reviewing training queue.
+
+* [admin reviews training queue]: to see who's waiting and how long.
+  - Admin goes to {training queue display}.
+  - Admin selects filtering and sorting criteria (if any).
+  - Admin pages through display showing both <Training Application>s.
+
+## Setting up training course.
+
+* [admin creates training event]: and assigns people to it.
+  - Admin creates new <Event>.
+  - Admin goes to {training queue display}.
+  - Admin selects <Event>.
+  - Admin selects some <Training Application>s.
+  - AMY creates <Task>s to add all the selected <Person>s to the <Event>.
+  - AMY connects the <Training Application>s to the <Event>.
+  - AMY moves <Training Application>s to <Assigned Application List>.
+  - AMY provides single-click "send to all" email link to admin to inform trainees.
+  - AMY creates <Training Progress> record for each <Person>.
+
+## Administering training course applications.
+
+* [admin removes people from training event][a]: and discards applications
+  - Admin goes to {assigned training application display}.
+  - Admin selects <Training Application>s.
+  - Admin removes them from training event.
+  - AMY deletes associated <Task>s.
+  - AMY deletes <Training Application>s.
+
+* [admin removes people from training event][b]: but keeps applications
+  - Admin goes to {assigned training application display}.
+  - Admin selects <Training Application>s.
+  - Admin removes them from training event.
+  - AMY deletes associated <Task>s.
+  - AMY moves <Training Application>s back to <Training Application List>.
+
+## Reviewing training progress.
+
+* [admin views training progress]: to see who's done what.
+  - Admin views {training progress page}.
+  - Admin selects filtering and sorting criteria (if any).
+
+## Tracking submission of written work.
+
+* [trainee reports submission of written work]: such as pull request or comment on lesson.
+  - Trainee views their own {training progress}.
+  - Trainee enters URL of written work.
+  - AMY updates <Training Task> associated with <Person> and <Event>.
```

Comment:
I've written on the assumption that `<Training Progress>` is one-to-many with `<Training Task>`, but you may decide to keep everything in `<Training Progress>` - that would be simpler...